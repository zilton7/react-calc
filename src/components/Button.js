/* eslint-disable react/prop-types */
import React from 'react';

const Button = ({ name, handleClick }) => ({
  name, handleClick, wide, color,
}) {
  return (
    <>
      <button
        type="button"
        name={name}
        onClick={handleClick}
        style={{
          background: color || '#f5913e',
          width: wide ? '50%' : '25%',
        }}
      >
        {name}
      </button>
    </>
  );
}

export default Button;
