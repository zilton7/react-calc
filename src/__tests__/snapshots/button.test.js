import renderer from 'react-test-renderer';
import Button from '../../components/Button';

it('renders Button correctly', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});
