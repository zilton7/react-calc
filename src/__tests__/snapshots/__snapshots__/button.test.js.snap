// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button correctly 1`] = `
<button
  style={
    Object {
      "background": "#f5913e",
      "width": "25%",
    }
  }
  type="button"
/>
`;
